src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -416,11 +416,15 @@ public float GetSingle()
         internal float GetSingleWithQuotes()
         {
             ReadOnlySpan<byte> span = GetUnescapedSpan();
-            if (!TryGetSingleCore(out float value, span))
+
+            char numberFormat = JsonReaderHelper.GetFloatingPointStandardParseFormat(span);
+            if (Utf8Parser.TryParse(span, out float value, out int bytesConsumed, numberFormat)
```

Comment:
Did you consider using `Utf8JsonReader._numberFormat`?

Reply:
Because the token is `JsonTokenType.String`, we don't go down the code path that would set it correctly: https://github.com/dotnet/runtime/blob/bfb01c942038e85e9853764d8f8940f9acb025e3/src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs#L1496

This causes a bug where a quoted number may not be parsed correctly because a previous JSON number (`JsonTokenType.Number`) set `_numberFormat` to `JsonConstants.ScientificNotationFormat`, and it was not updated on the next read for a number.

The bug fix is tested here (not explicitly, can do so in follow up): https://github.com/dotnet/runtime/blob/bfb01c942038e85e9853764d8f8940f9acb025e3/src/libraries/System.Text.Json/tests/Serialization/NumberHandlingTests.cs#L217-L221.

Reply:
It looks to me that `standardFormat` should not be provided since the default already allows scientific notation. 

For the current `_numberFormat` in `Utf8JsonReader`, it is supposed to be enabled only for numbers with scientific notation. If the code would stop providing `_numberFormat` to `Utf8Parser.GetSingle/Double/Decimal` we would get the same result so perhaps we should make an effort to remove that field from `Utf8JsonReader`.

With that said, I wouldn't like introducing more code that does unnecessary work BUT since the goal in this PR is to behave as much as we can as the reader, I guess is fine to keep it, but remove it later if said effort determines that `_numberFormat` can be safely removed.

Reply:
>  so perhaps we should make an effort to remove that field from Utf8JsonReader

We would need to benchmark/be sure that a slower code path is not taken for parsing numbers in scientific notation, if we don't specify the standard format.